qualtran/bloqs/block_encoding.py

```diff
@@ -182,69 +191,247 @@ def pretty_name(self) -> str:
+    a $(\alpha, a, \epsilon)$-block encoding of $H$ if it satisfies:
+
+    $$
+        \lVert H - \alpha (\langle G|_a\otimes I_s U |G\rangle_a \otimes I_s) \rVert
+        \le \epsilon,
+    $$
+
+    where $a$ is an ancilla register and $s$ is the system register, $U$ is a unitary sometimes
+    called a signal oracle, $\alpha \ge \lVert H\rVert$ (where $\lVert \cdot \rVert$
+    denotes the spectral norm), and $\epsilon$ is the precision to which the block
+    encoding is prepared. The state $|G\rangle_a$ is sometimes called the signal
+    state, and its form depends on the details of the block encoding.
+
+    For LCU based block encodings with $H = \sum_l w_l U_l$
+    we have
+    $$
+    U = \sum_l |l\rangle\langle l| \otimes U_l
+    $$
+    and $|G\rangle = \sum_l \sqrt{\frac{w_l}{\alpha}}|0\rangle_a$, which define the
+    usual SELECT and PREPARE oracles.
+
+    Other ways of building block encodings exist so we define the abstract base
+    class `BlockEncoding` bloq, which expects values for $\alpha$, $\epsilon$,
+    system and ancilla registers and a bloq which prepares the state $|G\rangle$.
+
+    Users must specify:
+    1. the normalization constant $\alpha \ge \lVert A \rVert$, where
+        $\lVert \cdot \rVert$ denotes the spectral norm.
+    2. the precision to which the block encoding is to be prepared ($\epsilon$).
+
+    Developers must provide a method to return a bloq to prepare $|G\rangle$.
+
+    References:
+        [Hamiltonian Simulation by Qubitization](https://quantum-journal.org/papers/q-2019-07-12-163/)
+            Sec 2 and 3 for introduction and definition of terms.
+
+        [The power of block-encoded matrix powers: improved regression techniques via faster Hamiltonian simulation](https://arxiv.org/abs/1804.01973)
+            Definition 3 page 8.
+
+    """
+
+    def pretty_name(self) -> str:
+        return 'B[H]'
+
+    @property
+    @abc.abstractmethod
+    def selection_registers(self) -> Tuple[Register, ...]:
+        """The ancilla registers `a` above."""
+
+    @property
+    @abc.abstractmethod
+    def junk_registers(self) -> Tuple[Register, ...]:
+        """Any additional junk registers not included in selection registers."""
+
+    @property
+    @abc.abstractmethod
+    def target_registers(self) -> Tuple[Register, ...]:
+        """The system registers of combined size `s`."""
+
+    @property
+    @abc.abstractmethod
+    def signal_state(self) -> PrepareOracle:
+        r"""Returns the signal / ancilla flag state $|G\rangle."""
+
+
 @attrs.frozen
-class BlackBoxBlockEncoding(Bloq):
-    r"""Black box implementation of a block encoding using SELECT and PREPARE.
+class LCUBlockEncoding(BlockEncoding):
+    r"""LCU based block encoding using SELECT and PREPARE oracles.
 
     Builds the block encoding via
     $$
-        \mathcal{B}[V] = \mathrm{PREPARE}^\dagger \cdot \mathrm{SELECT} \cdot \mathrm{PREPARE},
+        B[H] = \mathrm{SELECT}
```

Comment:
you don't really introduce the `B[H]` notation in the `BlockEncoding` docstring

Reply:
should probably use the words "linear combination of unitaries" somewhere in the docstring

Reply:
Done.